static/js/settings.js

```diff
@@ -382,5 +388,68 @@ var SettingsScreen = {
     }).catch(function() {
       updateNow.textContent = 'Error';
     });
+  },
+
+  showAuthorizationSettings: function() {
+    this.authorizationSettings.classList.remove('hidden');
+
+    fetch('/authorizations', {
+      headers: API.headers()
+    }).then(function (response) {
+      return response.json();
+    }).then(clients => {
+      const authorizationsList = document.getElementById('authorizations');
+      const noAuthorizations = document.getElementById('no-authorizations');
+      authorizationsList.innerHTML = '';
+
+      if (clients.length === 0) {
+        noAuthorizations.classList.remove('hidden');
+        authorizationsList.classList.add('hidden');
+        return;
+      }
+
+      noAuthorizations.classList.add('hidden');
+      authorizationsList.classList.remove('hidden');
+
+      for (let client of clients) {
+        let authorization = document.createElement('li');
+        authorization.classList.add('authorization-item');
+        let name = document.createElement('p');
+        name.textContent = client.name;
+        let origin = document.createElement('p');
+        origin.classList.add('origin');
+        origin.textContent = new URL(client.redirect_uri).host;
+        let revoke = document.createElement('input');
+        revoke.classList.add('revoke-button', 'text-button');
+        revoke.type = 'button';
+        revoke.value = 'Revoke';
+        revoke.dataset.clientId = client.id;
+        revoke.addEventListener('click', this.revokeAuthorization);
+
+        authorization.appendChild(name);
+        authorization.appendChild(origin);
+        authorization.appendChild(revoke);
+        authorizationsList.appendChild(authorization);
+      }
+    });
+  },
+
+  revokeAuthorization: function(event) {
+    let revoke = event.target;
+    if (!revoke.dataset.clientId) {
+      console.warn('Missing clientId on revoke', revoke);
+      return;
+    }
+    let clientId = revoke.dataset.clientId;
+
+    fetch('/authorizations/' + clientId, {
+      headers: API.headers(),
+      method: 'DELETE'
+    }).then(() => {
+      let authorization = revoke.parentNode;
+      authorization.parentNode.removeChild(authorization);
+    }).catch(err => {
+      console.warn('Unable to revoke', err);
+    });
```

Comment:
I noticed that when I revoke the last authorization, the screen just comes back blank rather than saying "No authorizations"

Reply:
Good catch! I'll fix this behavior